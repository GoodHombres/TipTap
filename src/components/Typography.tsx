import { Text, type TextProps } from "react-native";
import { cn } from "#helpers/cn";
import type { ObjectValues } from "#types/ObjectValues";

const Variant = {
  body: "body",
  link: "link",
  title: "title",
  subtitle: "subtitle",
} as const;

type Variant = ObjectValues<typeof Variant>;

const VariantClasses = {
  [Variant.body]: "text-base leading-[24px]",
  [Variant.link]: "text-base leading-[30px] text-primary",
  [Variant.title]: "text-2xl font-bold leading-[32px]",
  [Variant.subtitle]: "text-xl font-bold",
} satisfies Record<Variant, string>;

export type Props = TextProps & {
  variant?: Variant;
};

export function Typography({
  className,
  variant = Variant.body,
  ...rest
}: Props) {
  return (
    <Text
      className={cn([
        "text-foreground font-sans",
        VariantClasses[variant],
        className,
      ])}
      {...rest}
    />
  );
}
