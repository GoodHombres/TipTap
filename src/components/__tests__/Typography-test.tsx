import * as React from "react";
import renderer from "react-test-renderer";

import { Typography } from "../Typography";

it(`renders correctly`, () => {
  const tree = renderer
    .create(<Typography variant="body">Snapshot test!</Typography>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
