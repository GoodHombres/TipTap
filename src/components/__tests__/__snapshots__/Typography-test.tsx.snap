// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  className="text-base leading-[24px]"
>
  Snapshot test!
</Text>
`;
